src/Http/Routing/src/ValidationEndpointFilterFactory.cs

```diff
@@ -55,18 +69,20 @@ public static EndpointFilterDelegate Create(EndpointFilterFactoryContext context
         {
             ValidateContext? validateContext = null;
 
-            for (var i = 0; i < context.Arguments.Count; i++)
+            foreach (var entry in validatableParameters)
             {
-                var validatableParameter = validatableParameters[i];
-                var displayName = parameterDisplayNames[i];
+                if (entry.Index >= context.Arguments.Count)
+                {
+                    continue;
```

Comment:
```suggestion
                    break;
```
? 
In the next iteration then `entry.Index` will be even higher, so that condition will be true again, so exit early instead.

Reply:
```suggestion
                    break;
```

You can break early here instead of continuing since the list is built in ascending order.